tests/testsuite/cfg.rs

```diff
@@ -515,9 +515,73 @@ fn exclusive_dep_kinds() {
         .with_status(101)
         // can't find crate for `bar`
         .with_stderr_data(str![[r#"
+[WARNING] [ROOT]/foo/Cargo.toml: unexpected `cfg` condition name: `abc` in `[target.'cfg(abc)'.dependencies]`
+[WARNING] [ROOT]/foo/Cargo.toml: unexpected `cfg` condition name: `abc` in `[target.'cfg(not(abc))'.dependencies]`
 [COMPILING] foo v0.1.0 ([ROOT]/foo)
 error[E0463]: can't find crate for `bar`
 ...
 "#]])
         .run();
 }
+
+#[cargo_test]
+fn unexpected_cfgs_target() {
+    let p = project()
```

Comment:
Any new tests like this should be added in commits before the behavior change and that commit should pass.